### PR TITLE
Added mechanisms for retrieving tags by position within a sequence

### PR DESCRIPTION
**Pull request by ctb**
This is primitive functionality enabling certain sorts of queries into the graph from within Python. /cc @camillescott; at some point we should chat more about how this intersects with labeling.

### CODE REVIEW

**Comment by ctb** [comment]
(a review would be welcome, but not yet ready for merge.)

**Comment by ctb** [comment]
- [ ] Is it mergable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage.
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

**Comment by ctb** [comment]
Ready for review @luizirber @camillescott @mr-c

File: khmer/_khmermodule.cc [hunk 1]
```diff
@@ -1326,6 +1326,91 @@ static PyObject * hash_consume_and_tag(PyObject * self, PyObject * args)
...
+    khmer_KCountingHashObject * me = (khmer_KCountingHashObject *) self;
+    CountingHash * counting = me->counting;
+
+    const char * kmer_s = NULL;
+
+    if (!PyArg_ParseTuple(args, "s", &kmer_s)) {
+        return NULL;
+    }
+
+    if (strlen(kmer_s) < counting->ksize()) { // @@
```
Comment (luizirber) [inline khmer/_khmermodule.cc]: What is this comment?

Comment (ctb) [inline khmer/_khmermodule.cc]: Removed.

File: khmer/_khmermodule.cc [hunk 2]
```diff
@@ -1326,6 +1326,91 @@ static PyObject * hash_consume_and_tag(PyObject * self, PyObject * args)
...
+                         "starting kmer is smaller than the K size of the counting table");
+        return NULL;
+    }
+
+    SeenSet tags;
+
+    Py_BEGIN_ALLOW_THREADS
+
+    HashIntoType kmer, kmer_f, kmer_r;
+    kmer = _hash(kmer_s, counting->ksize(), kmer_f, kmer_r);
```
Comment (luizirber) [inline khmer/_khmermodule.cc]: GCC [complained][1] about kmer being defined but not used in this function, do you see it being used in the future? If not maybe just ignore the return value and remove it from the previous line?

[1]: http://ci.ged.msu.edu/job/khmer-pullrequest/1134/label=linux/warnings23Result/new/

**Comment by ctb** [comment]
Thanks @luizirber - ready for re-review.

**Comment by luizirber** [comment]
LGTM

File: khmer/_khmermodule.cc [hunk 3]
```diff
@@ -1326,6 +1326,91 @@ static PyObject * hash_consume_and_tag(PyObject * self, PyObject * args)
...
+            if (set_contains(counting->all_tags, kmer)) {
+                 posns.push_back(pos);
+                 tags.push_back(kmer);
+            }
+            pos++;
+        }
+    } catch (_khmer_signal &e) {
+        PyErr_SetString(PyExc_ValueError, e.get_message().c_str());
+        return NULL;
+    }
```
Comment (mr-c) [inline khmer/_khmermodule.cc]: 1356-1357 lack testing coverage and won't be called ever anyhow

Comment (ctb) [inline khmer/_khmermodule.cc]: fixed!

On Tue, Feb 10, 2015 at 11:16:55AM -0800, Michael R. Crusoe wrote:
> > +    try {
> > +        unsigned int pos = 1;
> > +        KMerIterator kmers(seq, counting->ksize());
> > +
> > +        while (!kmers.done()) {
> > +            HashIntoType kmer = kmers.next();
> > +            if (set_contains(counting->all_tags, kmer)) {
> > +                 posns.push_back(pos);
> > +                 tags.push_back(kmer);
> > +            }
> > +            pos++;
> > +        }
> > +    } catch (_khmer_signal &e) {
> > +        PyErr_SetString(PyExc_ValueError, e.get_message().c_str());
> > +        return NULL;
> > +    }
> 
> 1356-1357 lack testing coverage and won't be called ever anyhow
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/ged-lab/khmer/pull/749/files#r24441205
-- 
C. Titus Brown, ctbrown@ucdavis.edu

File: khmer/_khmermodule.cc [hunk 4]
```diff
@@ -1326,6 +1326,91 @@ static PyObject * hash_consume_and_tag(PyObject * self, PyObject * args)
...
+    const char * kmer_s = NULL;
+
+    if (!PyArg_ParseTuple(args, "s", &kmer_s)) {
+        return NULL;
+    }
+
+    if (strlen(kmer_s) < counting->ksize()) {
+        PyErr_SetString( PyExc_ValueError,
+                         "starting kmer is smaller than the K size of the counting table");
+        return NULL;
```
Comment (mr-c) [inline khmer/_khmermodule.cc]: This code branch is untested

File: khmer/_khmermodule.cc (hunk 3 above)
Comment (mr-c) [inline khmer/_khmermodule.cc]: Thanks; making new pull request

**Pull request merged**